recipes/daw_header_libraries/all/conanfile.py

```diff
@@ -1,71 +1,70 @@
-        "Visual Studio": "16",
-        "gcc": "8",
-        "clang": "7",
-        "apple-clang": "12.0",
-    }
-
     @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def _compiler_required_cpp17(self):
+        return {
+            "Visual Studio": "16",
+            "msvc": "14.4",
+            "gcc": "8",
+            "clang": "7",
+            "apple-clang": "12.0",
+        }
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def package_id(self):
+        self.info.clear()
 
     def validate(self):
         if self.settings.get_safe("compiler.cppstd"):
-            tools.check_min_cppstd(self, "17")
+            check_min_cppstd(self, "17")
 
-        minimum_version = self._compiler_required_cpp17.get(str(self.settings.compiler), False)
-        if minimum_version:
-            if tools.Version(self.settings.compiler.version) < minimum_version:
-                raise ConanInvalidConfiguration("{} requires C++17, which your compiler does not support.".format(self.name))
+        minimum_version = self._compiler_required_cpp17.get(str(self.info.settings.compiler), False)
+        if minimum_version and Version(self.info.settings.get_safe("compiler.version")) < minimum_version:
+            raise ConanInvalidConfiguration("{} requires C++17, which your compiler does not support.".format(self.name))
         else:
             self.output.warn("{0} requires C++17. Your compiler is unknown. Assuming it supports C++17.".format(self.name))
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
 
-    def _configure_cmake(self):
-        cmake = CMake(self)
-        cmake.configure(source_folder=self._source_subfolder)
-        return cmake
+    def build(self):
+        pass
 
     def package(self):
-        self.copy("LICENSE*", "licenses", self._source_subfolder)
-
-        cmake = self._configure_cmake()
-        cmake.install()
-
-        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
-
-    def package_id(self):
-        self.info.header_only()
+        copy(self, pattern="LICENSE*", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        copy(self, pattern="*.h", dst=os.path.join(self.package_folder, "include"), src=os.path.join(self.source_folder, "include"))
 
     def package_info(self):
-        self.cpp_info.filenames["cmake_find_package"] = "daw-header-libraries"
-        self.cpp_info.filenames["cmake_find_package_multi"] = "daw-header-libraries"
         self.cpp_info.set_property("cmake_file_name", "daw-header-libraries")
+        self.cpp_info.set_property("cmake_target_name", "daw::daw-header-libraries")
+        self.cpp_info.components["daw"].set_property("cmake_target_name", "daw::daw-header-libraries")
```

Comment:
to move after the TODO, this component is a side effect of cmake_find_package, so it will be removed in conan v2.

Reply:
Thank you for comment.
Fixed.